core-tests/src/test/java/org/mule/runtime/core/internal/construct/DefaultFlowTestCase.java

```diff
@@ -503,7 +520,30 @@ public void doNotExecuteOnErrorContinueDefinedOutsideTheFlow() throws MuleExcept
     assertThat(nonExpectedError.get(), is(false));
   }
 
-  private void startFlow() throws InitialisationException, MuleException {
+  @Test
+  @Issue("MULE-18873")
+  public void flowInsideProcessWithChildContextMustNotDropEvents()
+      throws MuleException, ExecutionException, InterruptedException {
+    CoreEvent testEvent = testEvent();
+    flow = (DefaultFlow) Flow.builder(FLOW_NAME, muleContext)
+        .source(flow.getSource())
+        .processors(singletonList(new BlockMessageProcessor()))
+        .processingStrategyFactory(new ProactorStreamEmitterProcessingStrategyFactory())
+        .build();
+    startFlow();
+    Flux<CoreEvent> flowProcessing = Flux
+        .from(MessageProcessors.processWithChildContext(testEvent, flow, Optional.of(from(FLOW_NAME))));
+    Future validation = executor.submit(() -> StepVerifier.create(flowProcessing)
```

Comment:
Is the executor needed? What do you think about directly this?
```
StepVerifier.create(flowProcessing)
        .expectNext(testEvent)
        .expectComplete()
        .verifyThenAssertThat()
        .hasNotDroppedElements();
```

Reply:
No needed at all (was a residue of trying to see if a latch could be used)